### ISSUE TITLE
Try using the E6 module structure to modularize code

### ISSUE THREAD

**Issue by markcornwell** [markcornwell/spacewar#23]
https://tylermcginnis.com/javascript-modules-iifes-commonjs-esmodules/

Things will get easier to manage if we apy attention to the module structure.  I think the matrix and vector operations really cry out to be in a separate module.  Also some the the drawing and rendering operations, and collision detection.  These are likely to be replaced by libraries in the future as we evolve.  Not committed to doing that just yet, but will keep our options open.

**Comment by markcornwell** [comment]
Got this working after some work.  Key gotcha's:
- need to load through http-server not direct from file
- work around into install chrome web server extension.  Configure through Chrome Apps button
- point the server directory at /home/mark/spacewar
- add type="module" to the http file
- careful about the import syntax
- this.foo = value no longer works at top level use  var foo = value or const foo = value
seems to be because no top level objects defined.  After I got this running it looks like a win.  The module structure makes for a much cleaner separation between modules and aids in makeing a comprehensive structure.

While I was at it, I factored out some of the self test code.  We now have the start of a simple test framework, that could grow.  Feeling good about the direction this takes us.  Better sect to tackle what lays ahead!
